common/configuration/manager.go

```diff
@@ -139,6 +138,20 @@ func (m *Manager[R]) SetJSON(ctx context.Context, pkey string, ref Ref, value js
 	return m.router.Set(ctx, ref, key)
 }
 
+// selectProvider selects the Provider using the supplied key or defaults to the sole provider
+// when no key is supplied and there exists exactly one provider.
+func (m *Manager[R]) selectProvider(key string) (Provider[R], error) {
```

Comment:
There's been a slight misunderstanding. This change should allow the controller to support multiple providers for reading but only one for writing, as usual, but the controller is the one that selects the writer at startup. This would also remove the types for setting the provider over gRPC. The client will still need these flags for the local case, but for the gRPC path it would be able to skip requiring the user to set a writer.

This isn't a high priority though, so let's park this one for now.

Reply:
sounds good I'll park this change